File: examples/javascript/prepare-subject-delete.mjs
import { memside } from "./memside-client.mjs";

const subjectId = process.argv[2];
if (!subjectId) {
  throw new Error("Usage: node prepare-subject-delete.mjs <subject-id>");
}

const preview = await memside(
  `/subjects/${encodeURIComponent(subjectId)}/deletion/prepare`,
  { method: "POST" }
);

console.log(JSON.stringify(preview, null, 2));

File: examples/python/prepare_subject_delete.py
import json
import sys
from urllib.parse import quote

from memside_http import api_request


if len(sys.argv) != 2:
    raise RuntimeError(
        "Usage: python prepare_subject_delete.py <subject-id>"
    )

preview = api_request(
    "POST",
    f"/subjects/{quote(sys.argv[1], safe='')}/deletion/prepare",
)

print(json.dumps(preview, indent=2))
